pkg/operator/ceph/nfs/controller_test.go

```diff
@@ -116,113 +213,260 @@ func TestCephNFSController(t *testing.T) {
+			assert.NoError(t, err)
+			assert.False(t, res.Requeue)
+			assertCephNFSReady(t, r)
+			assertResourcesExist(t, cCtx, "rook-ceph-nfs-my-nfs-a", "rook-ceph-nfs-my-nfs-b", "rook-ceph-nfs-my-nfs-c")
+			assert.Len(t, *deploymentsUpdated, 3)
+			assert.Equal(t, "rook-ceph-nfs-my-nfs-a", (*deploymentsUpdated)[0].Name)
+			assert.Equal(t, "rook-ceph-nfs-my-nfs-b", (*deploymentsUpdated)[1].Name)
+			assert.Equal(t, "rook-ceph-nfs-my-nfs-c", (*deploymentsUpdated)[2].Name)
+		})
+	})
+
+	t.Run("scale down nfs servers", func(t *testing.T) {
+		t.Run("scale from 3 to 2 servers", func(t *testing.T) {
+			cCtx := newContext(successExecutor(t))
+			nfs := baseCephNFS()
+			nfs.Spec.Server.Active = 3
+			cl := newControllerClient(nfs, cephClusterReady(cCtx))
+			r := newReconcile(cCtx, cl)
+
+			var deploymentsUpdated *[]*apps.Deployment
+			updateDeploymentAndWait, deploymentsUpdated = testopk8s.UpdateDeploymentAndWaitStub()
+
+			res, err := r.Reconcile(ctx, req)
+			assert.NoError(t, err)
+			assert.False(t, res.Requeue)
+			assertCephNFSReady(t, r)
+			assertResourcesExist(t, cCtx, "rook-ceph-nfs-my-nfs-a", "rook-ceph-nfs-my-nfs-b", "rook-ceph-nfs-my-nfs-c")
+
+			err = cl.Get(ctx, client.ObjectKeyFromObject(nfs), nfs)
+			assert.NoError(t, err)
+			nfs.Spec.Server.Active = 2
+			err = cl.Update(ctx, nfs, &client.UpdateOptions{})
+			assert.NoError(t, err)
+
+			res, err = r.Reconcile(ctx, req)
+			assert.NoError(t, err)
+			assert.False(t, res.Requeue)
+			assertCephNFSReady(t, r)
+			assertResourcesExist(t, cCtx, "rook-ceph-nfs-my-nfs-a", "rook-ceph-nfs-my-nfs-b")
+			assert.Len(t, *deploymentsUpdated, 2)
+			assert.Equal(t, "rook-ceph-nfs-my-nfs-a", (*deploymentsUpdated)[0].Name)
+			assert.Equal(t, "rook-ceph-nfs-my-nfs-b", (*deploymentsUpdated)[1].Name)
+		})
+
+		t.Run("scale from 3 to 1 servers", func(t *testing.T) {
+			cCtx := newContext(successExecutor(t))
+			nfs := baseCephNFS()
+			nfs.Spec.Server.Active = 3
+			cl := newControllerClient(nfs, cephClusterReady(cCtx))
+			r := newReconcile(cCtx, cl)
+
+			var deploymentsUpdated *[]*apps.Deployment
+			updateDeploymentAndWait, deploymentsUpdated = testopk8s.UpdateDeploymentAndWaitStub()
+
+			res, err := r.Reconcile(ctx, req)
+			assert.NoError(t, err)
+			assert.False(t, res.Requeue)
+			assertCephNFSReady(t, r)
+			assertResourcesExist(t, cCtx, "rook-ceph-nfs-my-nfs-a", "rook-ceph-nfs-my-nfs-b", "rook-ceph-nfs-my-nfs-c")
+
+			err = cl.Get(ctx, client.ObjectKeyFromObject(nfs), nfs)
+			assert.NoError(t, err)
+			nfs.Spec.Server.Active = 1
+			err = cl.Update(ctx, nfs, &client.UpdateOptions{})
+			assert.NoError(t, err)
+
+			res, err = r.Reconcile(ctx, req)
+			assert.NoError(t, err)
+			assert.False(t, res.Requeue)
+			assertCephNFSReady(t, r)
+			assertResourcesExist(t, cCtx, "rook-ceph-nfs-my-nfs-a")
+			assert.Len(t, *deploymentsUpdated, 1)
+			assert.Equal(t, "rook-ceph-nfs-my-nfs-a", (*deploymentsUpdated)[0].Name)
+		})
+	})
+
+	t.Run("multiple CephNFS clusters", func(t *testing.T) {
```

Comment:
Make sure we can run multiple and that downscaling one doesn't downscale another.